cartographer_ros/cartographer_ros/start_trajectory_main.cc

```diff
@@ -49,7 +51,15 @@ TrajectoryOptions LoadOptions() {
   auto lua_parameter_dictionary =
       cartographer::common::LuaParameterDictionary::NonReferenceCounted(
           code, std::move(file_resolver));
-  return CreateTrajectoryOptions(lua_parameter_dictionary.get());
+  if (!FLAGS_initial_pose.empty()) {
+    auto initial_trajectory_pose =
+        cartographer::common::LuaParameterDictionary::NonReferenceCounted(
+            "return " + FLAGS_initial_pose, std::move(file_resolver));
```

Comment:
Are you not moving std::move(file_resolver) a second time (see l. 53)?

Reply:
ah yes you are right. I wasn't sure how to handle here since LuaParameterDictionary isn't constructable without file resolver. Should I pass nullptr?

Reply:
I know you are not using the `file_resolver` and `nullptr` would not crash but please just construct another `file_resolver` inside the if-block and std::move it into your LuaParameterDictionary.

Please run this code once with and w/o flag to make sure it works as expeceted.

Reply:
I have confirmed it is working for both cases. And added file resolver for initial pose lua dict.